packages/dds/tree/src/simple-tree/core/withType.ts

```diff
@@ -14,19 +16,44 @@
  * Instead construct a real node of the desired type using its constructor.
  * @privateRemarks
  * This prevents non-nodes from being accidentally used as nodes, as well as allows the type checker to distinguish different node types.
- * @public
+ * @deprecated External code should use `Tree.schema(theNode)` for runtime data access, and for typechecking and internally {@link typeSchemaSymbol} provides a superset of this functionality.
+ * @system @public
  */
 export const typeNameSymbol: unique symbol = Symbol("TreeNode Type");
 
+/**
+ * The type of a {@link TreeNode}.
+ * For more information about the type, use `Tree.schema(theNode)` instead.
+ * @remarks
+ * This symbol mainly exists on nodes to allow TypeScript to provide more accurate type checking.
+ * `Tree.is` and `Tree.schema` provide a superset of this information in more friendly ways.
+ *
+ * This symbol should not manually be added to objects as doing so allows the object to be invalidly used where nodes are expected.
+ * Instead construct a real node of the desired type using its constructor.
+ * @privateRemarks
+ * This prevents non-nodes from being accidentally used as nodes, as well as allows the type checker to distinguish different node types.
+ * @system @public
+ */
+export const typeSchemaSymbol: unique symbol = Symbol("TreeNode Schema");
+
 /**
  * Adds a type symbol to a type for stronger typing.
  * @remarks
  * An implementation detail of {@link TreeNode}'s strong typing setup: not intended for direct use outside of this package.
  * @sealed @public
  */
-export interface WithType<TName extends string = string> {
+export interface WithType<
+	out Name extends string = string,
```

Comment:
Some new-ish guidance around type parameter naming that is relevant here: https://github.com/microsoft/FluidFramework/wiki/Coding-Guidelines#-do-prefix-type-parameters-with-t

Reply:
We should probably add `@typeParam` comments for the two type parameters as well.

Reply:
Done.